packages/solid-crs-webid/Dockerfile

```diff
@@ -23,7 +24,8 @@ RUN npm install  --unsafe-perm
 COPY . /community-server
 
 # Copy initial data
-# TODO add .acl to root for proxy access
+COPY ./data/$ENVIRONMENT/ /data
+COPY ./config/ /config
```

Comment:
Only `/data` (not `./data`) is [mounted as a volume](https://github.com/netwerk-digitaal-erfgoed/infrastructure/blob/master/k8s/solid-crs/solid-crs-pods-stateful-set.yml#L23-L25), `/config` is created by the `Dockerfile`.

> The ./config/ directory contains an edited version of the default file.json (account registration is disabled here)

This should work.

> The ./data/$ENVIRONMENT/ directory contains resources that manage access for the id-proxy to the root of the pod server, so that it can write new webid documents. It also contains the profile document for the proxy itself.

This won't work because the volume is mounted _over_ the contents of `/data` included by the Docker build, so these original contents will disappear.

A better way then might be to have the app check for and write (if necessary) that data on boot.

Reply:
> This won't work because the volume is mounted _over_ the contents of `/data` included by the Docker build, so these original contents will disappear.
> 
> A better way then might be to have the app check for and create/update (if necessary) that data on boot.

Can this be done when starting the container as part of the start command (perhaps in the form of a simple .sh script)? I'd like to avoid making code changes for the pod server if that is what you mean with "the app".

Reply:
By ‘the app’ I mean the `solid-crs-webid` package. The question is: is `data/local` an application or an infrastructure concern? I suggest the former, because the app [already copies that data on startup](https://github.com/netwerk-digitaal-erfgoed/solid-crs/pull/538/files#diff-5dddeee4a033efa0f78fd6bf3fae6da3acc822eae0022ab65dc0c9982d293f2dR8). The data is checked into the app repository, and if it changes, it should be updates in the container, too.

My suggestion, then, is to re-use the copy-on-start mechanism by making the target directory (currently hard-coded `/tmp`) overridable:

- either by using env vars (preferable);
- or by adding a `start-prod` that copies to `/data` instead, which is the path on production.